File: src/main/java/nl/jboi/minecraft/bytecart/ModifiableRunnable.java
package nl.jboi.minecraft.bytecart;

/**
 * Represents a runnable that can updates its internal parameter
 *
 * @param <T> Type of the internal parameter
 */
public interface ModifiableRunnable<T> extends Runnable {

    /**
     * Updates the parameter variable
     *
     * @param t Object to be set as parameter
     */
    void SetParam(T t);
}

File: src/main/kotlin/nl/jboi/minecraft/bytecart/util/ModifiableRunnable.kt
package nl.jboi.minecraft.bytecart.util

/**
 * Represents a runnable that can updates its internal parameter
 *
 * @param T Type of the internal parameter
 */
interface ModifiableRunnable<T> : Runnable {
    /**
     * Updates the parameter variable
     *
     * @param t Object to be set as parameter
     */
    fun setParam(t: T)
}